### PR TITLE
Feature/store levels

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Store the level and experience per job
- Reduce pirate drops
- Fix block dupe bug with racks

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java [hunk 1]
```diff
@@ -279,9 +295,9 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
             isAsleep = compound.getBoolean(TAG_ASLEEP);
         }
 
-        if (level > MAX_CITIZEN_LEVEL)
+        if (job != null && levels.getOrDefault(job.getName(), 0) > MAX_CITIZEN_LEVEL)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/CitizenData.java]: shouldnt this check be in the for loop reading the value? To check the Max of every value, not just the current jobs

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java [hunk 2]
```diff
@@ -176,15 +172,15 @@
...
+    private Map<String, Integer> levels =  new HashMap<>();
 
     /**
-     * The total amount of experience the citizen has.
-     * This also includes the amount of experience within their Experience Bar.
+     * The total amount of experiences the citizen has depending on his job.
+     * This also includes the amount of experiences within their Experience Bar.
      */
-    private double experience;
+    private Map<String, Double> experiences =  new HashMap<>();
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/CitizenData.java]: may be easier to combine it in one HashMap<String, Tuple<Integer,Double>> ? And as far as I can see the hashmaps can be final

**Comment by someaddons** [comment]
Looks fine to me apart from those small things

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/CitizenData.java]: This is actually only a hotfix to reset workers which had a too high level before we added the level cap. That's something we can delete in the future again.

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/CitizenData.java]: yeah

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/CitizenData.java]: well if we were to ever change the constant again you'd need it :D always good to validate the data upon rereading^^

**Pull request merged**